### PR TITLE
backport math.*integer from 5.3

### PR DESCRIPTION
**Pull request by fperrad**
same behavior (and limitations) as [lua-compat53](https://github.com/keplerproject/lua-compat-5.3)

### CODE REVIEW

**Comment by fperrad** [comment]
doc and smoke tests added

File: src/lib_math.c [hunk 1]
```diff
@@ -96,6 +96,51 @@ LJLIB_ASM_(math_max)		LJLIB_REC(math_minmax IR_MAX)
 
 LJLIB_PUSH(3.14159265358979323846) LJLIB_SET(pi)
 LJLIB_PUSH(1e310) LJLIB_SET(huge)
+LJLIB_PUSH(9007199254740992) LJLIB_SET(maxinteger)	/* 2^53 */
+LJLIB_PUSH(-9007199254740992) LJLIB_SET(mininteger)
```
Comment (siddhesh) [inline src/lib_math.c:100]: I suppose that's an unfortunate limitation of always storing numbers as float; lua is able to convert all the way up to `2^63-1` both ways when the input is an integer (i.e. `9223372036854775807` and not `9223372036854775807.0`), which is what it should be.  Please mention this difference in the documentation, hopefully we can fix this up in future.

File: src/lib_math.c [hunk 2]
```diff
@@ -96,6 +96,51 @@ LJLIB_ASM_(math_max)		LJLIB_REC(math_minmax IR_MAX)
...
+      return n
+    end
+    return nil
+  end
+*/
+
+LJLIB_LUA(math_type) /*
+  function(n)
+    if type(n) == 'number' then
+      if n <= 9007199254740992 and n >= -9007199254740992 and n % 1 == 0 then
```
Comment (siddhesh) [inline src/lib_math.c:114]: Likewise, please mention the limitation in the documentation.

**Review by siddhesh — CHANGES_REQUESTED** [review]
Looks OK overall, just needs a bit more documentation.  Please add a bit in the documentation describing how number handling is limited in luajit and hence these math functions can handle only 53-bit integers, unlike lua which can do the full 64-bit integers when specified as integers.

**Comment by fperrad** [comment]
add limitations in extensions.md

**Comment by joe-92** [comment]
I can imagine that the functions `math.tointeger`, `math.type` and `math.ult` are often used, so wouldn't make it sense to add them as fast functions?

**Comment by siddhesh** [comment]
The functions are compact enough that when jitted, they'll inline quite well and likely be more performant than C calls into fast functions.

**Review by siddhesh — APPROVED** [review]

**Pull request merged**